spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/ActuatorOperations.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.app;
+
+/**
+ * Deployers may implement this extension to invoke actuator endpoints of deployed app instances.
+ *
+ * @author David Turanski
+ */
+public interface ActuatorOperations {
+
+	/**
+	 * Get a resource from an actuator path.
+	 * @param deploymentId the deployment ID of the deployed app.
+	 * @param instanceIndex index of the app instance.
+	 * @param endpoint the endpoint path relative to the base actuator URL for the instance.
+	 * @param responseType the expected response type.
+	 * @return the contents as the given type.
+	 */
+	<T> T getFromActuator(String deploymentId, int instanceIndex, String endpoint, Class<T> responseType);
```

Comment:
Do we envision needing to [pass parameters](https://docs.spring.io/spring-boot/docs/2.4.8/reference/htmlsingle/#production-ready-endpoints-custom-input) to the endpoint?

Reply:
A couple of actuator endpoints support query parameters, but I don't consider them in scope for SCDF use cases. For example, the pre-packaged apps only expose health, info, and bindings.